### ISSUE TITLE
Shell: Here documents and Here strings

### ISSUE THREAD

**Issue by ADKaster** [SerenityOS/serenity#4283]
Rumor has it all the cool shells support heredocs and herestrings.

You can see one right here!
```sh
/mnt/c/Users/Andrew$ cat << EOF
> Hello, World!
> EOF
Hello, World!
```
But not in serenity :(
![heredoc](https://user-images.githubusercontent.com/8388494/100699894-62f10280-3359-11eb-8257-ee2bc0865f4c.PNG)

Dr. Posix:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_07_04
Mr Bash:
https://www.gnu.org/software/bash/manual/bash.html#Here-Documents

**Comment by alimpfard** [comment]
> be me, minding my own business
> see this issue
> "hmm, I wonder what the accepted behaviour regarding stuff after these heredocs is
> `echo << EOF 1 2 $(echo \`
> `foo)`
> `lol no print me`
> `EOF`
> "1 2 foo"
> okay, that makes a little bit of sense
> `cat <<eof1; cat <<eof2`
> `hello`
> `eof1`
> `friends`
> `eof2`
> "hello\nfriends"
> now you're just being weird

TL;DR it seems a little too special.
how many people actually use the second weirdly interleaved heredocs? sounds like a pain to get right

**Comment by alimpfard** [comment]
My initial thoughts on this after spending a little bit of time messing around with it on zsh and bash:
- uhhh...why? strings are already multiline, what's the point of these things?
- if `echo $some_string | your_command` looks bad, the thing that pipes strings (`command <<< 'some string'`) can be used to achieve the same effect...no?
- this:
```sh
if cat << EOF1; then cat << EOF2; else cat << EOF3; fi
well
EOF1
hello
EOF2
friends
EOF3
```
- guess what, these heredocs actually have to be in _just_ the right order!

my current thoughts on implementing this: 
- I'd rather just implement `<<< string`, more readable, less of a headache to use, less of a pain to implement
- if we _have_ to have heredocs, I'd rather treat them as another kind of multiline string instead of allowing that weirdo if statement above.

Though maybe I'm just not seeing the obvious and these actually have a purpose outside of confusing people?

**Comment by ADKaster** [comment]
Ahhhh I didn't realize that strings were already multiline. That takes most of the fun out of the feature.

I really only use them to do one of three things.

1) write out a short to medium length file in a script without using 
echo >> "\t\tThe next line\n" all over the place
2) script a moderately interesting set of actions for an ssh session.
3) create a really short file when I'm too lazy to even open vim to do it.

**Comment by ADKaster** [comment]
And to answer your earlier question, I didn't even know you could chain them slash have interleaved ones. That part of it definitely feels very clunky.
